golem-cli/tests/api_definition.rs

```diff
@@ -197,18 +196,25 @@ fn golem_def_with_response(
         id: id.to_string(),
         version: "0.1.0".to_string(),
         draft: true,
-        routes: vec![Route {
+        routes: vec![RouteData {
```

Comment:
RouteData is the user facing structure.
The module was so confusing with "same" names all over the places, with redundant serialize, deserialize.
So they are removed from the core data structures, and kept it only in API facing side